### DISCUSSION TITLE
LCE_compute_engine

### DISCUSSION THREAD

**Discussion by christophezeinaty** [larq/compute-engine#779] [Q&A] [answered]
Hello I am trying to integrate to my c++ code the larq compute engine using a cmake file, and I am new to this so I am failing to do that, I cant find any docs on that, I already managed to add tensorflow lite and I am doing normal inference with my c++ code, so my question is how can I add LCE using cmake file and should I remove the tensorflow lite that I added since tensorflow lite is part of the LCE build ?

**Comment by Tombana** [comment, marked as answer]
Hello,
If you are using tensorflow-lite with cmake, then you probably have something like this:
```cmake
add_subdirectory( <path_to_tensorflow_lite> "${CMAKE_CURRENT_BINARY_DIR}/tensorflow-lite" EXCLUDE_FROM_ALL )

...

target_link_libraries( <your_binary>  tensorflow-lite )
```

To use LCE's binary layers, do the following instead:

```cmake
add_subdirectory( <path_to_larq_compute_engine> )

...

target_link_libraries( <your_binary>  larq-compute-engine )
```
(remove the old `add_subdirectory` for tensorflow-lite, that is done inside the larq-compute-engine cmake file).

Note that you have to update compute-engine to the latest `main` branch because I just merged a fix for this (https://github.com/larq/compute-engine/pull/780).

**Reply by christophezeinaty** [reply]
Thank you it worked note that in the CMakeLists.txt of the compute-engine lib there is an issue with ` set(TENSORFLOW_SOURCE_DIR "${CMAKE_CURRENT_LIST_DIR}/third_party/tensorflow/")`  there is an extra "/" at the end which won't let you run the cmake .. just need to remove it and do a push on the main.

**Reply by Tombana** [reply]
Thanks!
